driver/src/main/com/mongodb/client/gridfs/GridFSBucketImpl.java

```diff
@@ -0,0 +1,321 @@
+    }
+
+    private MongoCollection<BsonDocument> getChunksCollection() {
+        return database.getCollection(bucketName + ".chunks", BsonDocument.class)
+                .withCodecRegistry(DEFAULT_CODEC_REGISTRY)
+                .withReadPreference(readPreference)
+                .withWriteConcern(writeConcern);
+    }
+
+    private void checkCreateIndex() {
+        if (!checkedIndexes) {
+            if (filesCollection.withReadPreference(primary()).find()
+                    .projection(new BsonDocument("_id", new BsonInt32(1))).first() == null) {
+                BsonDocument filesIndex = Indexes.ascending("filename", "uploadDate")
+                        .toBsonDocument(BsonDocument.class, DEFAULT_CODEC_REGISTRY);
+                if (!hasIndex(filesCollection, filesIndex)) {
+                    filesCollection.createIndex(filesIndex);
+                }
+                BsonDocument chunksIndex = Indexes.ascending("files_id", "n")
+                        .toBsonDocument(BsonDocument.class, DEFAULT_CODEC_REGISTRY);
+                if (!hasIndex(chunksCollection, chunksIndex)) {
+                    chunksCollection.createIndex(chunksIndex, new IndexOptions().unique(true));
+                }
+            }
+            checkedIndexes = true;
+        }
+    }
+
+    private boolean hasIndex(final MongoCollection<BsonDocument> collection, final BsonDocument index) {
+        boolean hasIndex = false;
+        ArrayList<BsonDocument> indexes = collection.listIndexes(BsonDocument.class)
+                .into(new ArrayList<BsonDocument>());
+        for (BsonDocument indexDoc : indexes) {
+            if (indexDoc.getDocument("key").equals(index)) {
+                hasIndex = true;
+                break;
+            }
+        }
+        return hasIndex;
+    }
+
+    private GridFSFile getFileByName(final String filename, final GridFSDownloadByNameOptions options) {
+        int revision = options.getRevision();
+        int skip;
+        BsonInt32 sort;
+        if (revision >= 0) {
+            skip = revision;
+            sort = new BsonInt32(1);
+        } else {
+            skip = (-revision) - 1;
+            sort = new BsonInt32(-1);
+        }
+
+        GridFSFile fileInfo = find(new BsonDocument("filename", new BsonString(filename)))
+                .skip(skip).sort(new BsonDocument("uploadDate", sort)).first();
+        if (fileInfo == null) {
+            throw new MongoGridFSException(format("No file found with the filename: %s and revision: %s", filename, revision));
+        }
+        return fileInfo;
+    }
+
+    private GridFSDownloadStream findTheFileInfoAndOpenDownloadStream(final BsonValue id) {
+        GridFSFile fileInfo = find(new BsonDocument("_id", id)).first();
+        if (fileInfo == null) {
+            throw new MongoGridFSException(format("No file found with the id: %s", id));
+        }
+        return new GridFSDownloadStreamImpl(fileInfo, chunksCollection);
```

Comment:
There is no open method on Java streams.  They are implicitly open on construction.